src/util/ParseableDuration.h

```diff
@@ -29,6 +43,18 @@ class ParseableDuration {
 
     is >> arg;
 
+    try {
+      result = ParseableDuration<DurationType>::fromString(arg);
+    } catch (const std::runtime_error&) {
+      // >> operator does not throw, it sets the error state
+      is.setstate(is.rdstate() | std::ios_base::failbit);
+    }
+
+    return is;
+  }
+
+  static ParseableDuration<DurationType> fromString(std::string_view arg) {
+    using namespace std::chrono;
     if (auto matcher = ctre::match<"(-?\\d+)(ns|us|ms|s|min|h)">(arg)) {
```

Comment:
Can the regex include some optional whictespace at the beginning, the end, and between the number and the datatype?